Subject: [PATCH] VolumeInfo: support to exfat fs.

set maxFileSize for exfat fs is 256TB. TODO, this value
maybe set to 16TB.

diff --git a/core/java/android/os/storage/VolumeInfo.java b/core/java/android/os/storage/VolumeInfo.java
index 19210b5..460ca17 100644
--- a/core/java/android/os/storage/VolumeInfo.java
+++ b/core/java/android/os/storage/VolumeInfo.java
@@ -385,6 +385,12 @@
                 maxFileSize = 4294967295L;
             }
 
+            if ("exfat".equals(fsType)) {
+                // Now, the max file size is 256TB.
+                // TODO: the max file size maybe support to 16EB.
+                maxFileSize = 256L * 1024 * 1024 * 1024 * 1024;
+            }
+
         } else {
             throw new IllegalStateException("Unexpected volume type " + type);
         }
